tools/cli/adminFailoverCommands.go

```diff
@@ -0,0 +1,262 @@
+	if err != nil {
+		ErrorAndExit("Failed to pause failover workflow", err)
+	}
+	fmt.Println("Failover paused")
+}
+
+// AdminFailoverResume resume a paused failover workflow
+func AdminFailoverResume(c *cli.Context) {
+	err := executePauseOrResume(c, false)
+	if err != nil {
+		ErrorAndExit("Failed to resume failover workflow", err)
+	}
+	fmt.Println("Failover resumed")
+}
+
+func executePauseOrResume(c *cli.Context, isPause bool) error {
+	client := getCadenceClient(c)
+	tcCtx, cancel := newContext(c)
+	defer cancel()
+
+	runID := getRunID(c)
+	var signalName string
+	if isPause {
+		signalName = failovermanager.PauseSignal
+	} else {
+		signalName = failovermanager.ResumeSignal
+	}
+
+	return client.SignalWorkflow(tcCtx, failovermanager.WorkflowID, runID, signalName, nil)
+}
+
+// AdminFailoverQuery query a failover workflow
+func AdminFailoverQuery(c *cli.Context) {
+	client := getCadenceClient(c)
+	tcCtx, cancel := newContext(c)
+	defer cancel()
+
+	runID := getRunID(c)
+
+	queryResult, err := client.QueryWorkflow(tcCtx, failovermanager.WorkflowID, runID, failovermanager.QueryType)
+	if err != nil {
+		ErrorAndExit("Failed to query failover workflow", err)
+	}
+	if !queryResult.HasValue() {
+		ErrorAndExit("QueryResult has no value", nil)
+	}
+
+	var result failovermanager.QueryResult
+	queryResult.Get(&result)
+
+	descResp, err := client.DescribeWorkflowExecution(tcCtx, failovermanager.WorkflowID, runID)
+	if err != nil {
+		ErrorAndExit("Failed to describe workflow", err)
+	}
+	if descResp.WorkflowExecutionInfo.CloseStatus.Equals(shared.WorkflowExecutionCloseStatusTerminated) {
+		result.State = failovermanager.WorkflowAborted
+	}
+	prettyPrintJSONObject(result)
+}
+
+// AdminFailoverAbort abort a failover workflow
+func AdminFailoverAbort(c *cli.Context) {
+	client := getCadenceClient(c)
+	tcCtx, cancel := newContext(c)
+	defer cancel()
+
+	reason := c.String(FlagReason)
+	if len(reason) == 0 {
+		reason = defaultAbortReason
+	}
+	runID := getRunID(c)
+
+	err := client.TerminateWorkflow(tcCtx, failovermanager.WorkflowID, runID, reason, nil)
+	if err != nil {
+		ErrorAndExit("Failed to abort failover workflow", err)
+	}
+
+	fmt.Println("Failover aborted")
+}
+
+// AdminFailoverRollback rollback a failover run
+func AdminFailoverRollback(c *cli.Context) {
+	client := getCadenceClient(c)
+	tcCtx, cancel := newContext(c)
+	defer cancel()
+
+	runID := getRunID(c)
+
+	queryResp, err := client.QueryWorkflow(tcCtx, failovermanager.WorkflowID, runID, failovermanager.QueryType)
+	if err != nil {
+		ErrorAndExit("Failed to query failover workflow", err)
+	}
+	if !queryResp.HasValue() {
+		ErrorAndExit("QueryResult has no value", nil)
+	}
+	var queryResult failovermanager.QueryResult
+	queryResp.Get(&queryResult)
+
+	if isWorkflowRunning(&queryResult) {
+		err := client.TerminateWorkflow(tcCtx, failovermanager.WorkflowID, runID, "Rollback", nil)
+		if err != nil {
+			ErrorAndExit("Failed to terminate failover workflow", err)
+		}
+	}
```

Comment:
Shall we query the workflow again after terminating it as the `SuccessDomains` may change in the middle?

Reply:
that's true, added.